src/test/msgr/CMakeLists.txt

```diff
@@ -16,15 +16,15 @@ add_executable(ceph_test_async_networkstack
   test_async_networkstack.cc
   $<TARGET_OBJECTS:unit-main>
   )
-target_link_libraries(ceph_test_async_networkstack global ${CRYPTO_LIBS} ${BLKID_LIBRARIES} ${CMAKE_DL_LIBS} ${UNITTEST_LIBS})
+target_link_libraries(ceph_test_async_networkstack global-static ${CRYPTO_LIBS} ${BLKID_LIBRARIES} ${CMAKE_DL_LIBS} ${UNITTEST_LIBS})
```

Comment:
how is ceph_test_async_networkstack linked against two copies of DPDK?

Reply:
target_link_libraries use the PUBLIC link property by default, Libraries following PUBLIC are linked to, and are made part of the link interface. check build/src/test/msgr/CMakeFiles/ceph_perf_msgr_client.dir/link.txt:
../../../lib/libceph-common.so ../../../lib/libjson_spirit.a ../../../lib/libcommon_    utf8.a ../../../lib/liberasure_code.a ../../../lib/libcrc32.a ../../../lib/libarch.a -lresolv ../../../boost/lib/libboost_thread.a ../../../boost/lib/libboost_chrono.a ../../../boo    st/lib/libboost_atomic.a ../../../boost/lib/libboost_system.a ../../../boost/lib/libboost_random.a ../../../boost/lib/libboost_program_options.a ../../../boost/lib/libboost_date_ti    me.a ../../../boost/lib/libboost_iostreams.a ../../../boost/lib/libboost_regex.a -lstdc++fs ../../../lib/libfmt.a /usr/lib/aarch64-linux-gnu/libblkid.so /usr/lib/aarch64-linux-gnu/    libcrypto.so /lib/aarch64-linux-gnu/libudev.so /usr/lib/aarch64-linux-gnu/libibverbs.so /usr/lib/aarch64-linux-gnu/librdmacm.so /usr/lib/aarch64-linux-gnu/libz.so ../../../lib/libc    ommon_async_dpdk.a -Wl,--whole-archive /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_bus_pci.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_cmdline.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_eal.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_ethdev.a /home/chunsong/ceph/obj-aarch64-linux-    gnu/src/dpdk/lib/librte_kvargs.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_mbuf.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_mempool.a /home/    chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_mempool_ring.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_pci.a /home/chunsong/ceph/obj-aarch64-linux-gnu    /src/dpdk/lib/librte_hash.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_net.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_pmd_i40e.a /home/chuns    ong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_pmd_ixgbe.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_pmd_hns3.a /home/chunsong/ceph/obj-aarch64-linux-gnu/sr    c/dpdk/lib/librte_pmd_hinic.a /home/chunsong/ceph/obj-aarch64-linux-gnu/src/dpdk/lib/librte_ring.a -Wl,--no-whole-archive  -Wl,-lnuma -Wl,-lpthread,-ldl ../../../lib/libgmock.a ../    ../../lib/libgtest.a -lpthread -ldl
it link libceph-common.so and librte_eal.a ,this cause two copies of eal